test/reflection.jl

```diff
@@ -274,6 +274,15 @@ end
 end
 @test functionloc(f15447)[2] > 0
 
+# issue #14346
+@noinline function f14346(id, mask, limit)
+    if id <= limit && mask[id]
+        return true
+    end
+end
+f14346(1, [true], 1)
+functionloc(f14346)
```

Comment:
it would be good to also `@test` these against the expected values (e.g. `@__LINE__ - 6`)